include/deal.II/non_matching/fe_values.h

```diff
@@ -368,6 +369,377 @@ namespace NonMatching
     DiscreteQuadratureGenerator<dim> quadrature_generator;
   };
 
+
+
+  /**
```

Comment:
Would you mind adding a small drawing so that it is clear what you are trying to achieve here? It took me a while to actually understand where you are integrating here. Admittedly, the name confused me (even if it is coherent with what we have in the library for the interfaces between cells):  I started reading this PR being convinced that you were providing a `FEInterfaceValues` like API for the `FEImmersedSurfaceValues` class (i.e., psi(x) = 0), and *not* for all other faces of the cell that are cut by the interface...

Reply:
> I started reading this PR being convinced that you were providing a FEInterfaceValues like API for the FEImmersedSurfaceValues class (i.e., psi(x) = 0),

@luca-heltai  Out of couriousity, how would that look like? Does this make sense in "continuous" case or do we need something like XFEM?